### PR TITLE
Cloudedit

### PR DESCRIPTION
**Pull request by aheifetz**
updates to make a single instance - need to re-think the unit tests.  The DSL can pass the unit then fail on CF create!

### CODE REVIEW

File: lib/ec2/Instance.js [hunk 1]
```diff
@@ -45,8 +46,41 @@ Instance = Resource.registerPropertyPrototypes(Instance, propertyMap);
...
+            }
+        }
+    
+    
+    var nic = {'AssociatePublicIpAddress' : true, 'DeviceIndex':'0','SubnetId':subnetId, 'GroupSet':GroupSet}
+    this.node.Properties.NetworkInterfaces.push(nic);
+    //TODO ensure no subnet Id on instance! 
+    return this;
+    
+}
```
Comment (owap) [inline lib/ec2/Instance.js]: Missing semicolon

File: lib/ec2/Instance.js [hunk 2]
```diff
@@ -45,8 +46,41 @@ Instance = Resource.registerPropertyPrototypes(Instance, propertyMap);
...
+
+
 Instance.prototype.name = function (name) {
-    this.node.Properties.Tags.push({'Key': 'Name', Value: name, 'PropagateAtLaunch': 'true'});
+    if( this.node.Properties.Tags ==null){
```
Comment (owap) [inline lib/ec2/Instance.js]: Expected '===' and instead saw '=='.

File: lib/ec2/Instance.js [hunk 3]
```diff
@@ -45,8 +46,41 @@ Instance = Resource.registerPropertyPrototypes(Instance, propertyMap);
...
+        }
+        // Automatically convert to references if the values are strings
+        for(var i=0; i<GroupSet.length; i++){
+            if(typeof GroupSet[i] === 'string'){
+                GroupSet[i] = { 'Ref': GroupSet[i] };
+            }
+        }
+    
+    
+    var nic = {'AssociatePublicIpAddress' : true, 'DeviceIndex':'0','SubnetId':subnetId, 'GroupSet':GroupSet}
```
Comment (owap) [inline lib/ec2/Instance.js]: Missing semicolon

File: lib/ec2/SecurityGroup.js [hunk 4]
```diff
@@ -27,6 +30,22 @@ var SecurityGroup = function (id, description) {
...
 
+///////////////////////////////////////
+// Security Group - specific Functions
+///////////////////////////////////////
+
+SecurityGroup.prototype.CidrIngressRule = function(cidrIP, to, from, protocal) {
+    var ingress = this.node.Properties.SecurityGroupIngress;
+    if (!ingress) {
+        ingress = this.node.Properties.SecurityGroupIngress = [];
+    };
```
Comment (owap) [inline lib/ec2/SecurityGroup.js]: Unnecessary semicolon

File: lib/ec2/SecurityGroup.js [hunk 5]
```diff
@@ -27,6 +30,22 @@ var SecurityGroup = function (id, description) {
...
+    var rule = {'CidrIp':cidrIP,'FromPort':from,'ToPort':to,'IpProtocol':protocal}
```
Comment (owap) [inline lib/ec2/SecurityGroup.js]: Missing semicolon

File: lib/ec2/SecurityGroup.js [hunk 6]
```diff
@@ -27,6 +30,22 @@ var SecurityGroup = function (id, description) {
...
+    ingress.push(rule);
+    return this;
+}
```
Comment (owap) [inline lib/ec2/SecurityGroup.js]: Missing semicolon

**Comment by owap** [comment]
In `Taggable.js`:
    + Line 35: missing semicolon
    + Line 22: need a `===` (in place of a `==`)

If you type `$ grunt` on in the command line, it will run JSHint and all the unit tests :)

File: lib/ec2/Instance.js (hunk 1 above)
Comment (aheifetz) [inline lib/ec2/Instance.js]: Agreed

File: lib/ec2/Instance.js (hunk 2 above)
Comment (aheifetz) [inline lib/ec2/Instance.js]: Agreed - switch back between java too much :)

**Comment by aheifetz** [comment]
thanks - making the changes after lunch!

**Comment by owap** [comment]
I've got the changes nearly ready to go -- don't sweat it :+1:

**Comment by aheifetz** [comment]
thanks for adding the fixes - should I self merge?

**Comment by owap** [comment]
You shouldn't ever self merge :wink:

**Comment by aheifetz** [comment]
And I should always re-test :)

**Pull request merged**